### PR TITLE
Finished assignment 1

### PR DESCRIPTION
**Pull request by djh2020**
2x solutions for the music sorting. 
zip, which is quick and iteritems() which was more readable, at least to me, but it was slower when I benchmarked it.
Could not get an iterator to act like range() i.e. be stateless.

### CODE REVIEW

File: students/darrell/wk1 - assignments/generator_solution.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    while True:
+        yield seq[-1]
+        if len(seq) == 1:
+            seq.append(1)
+        else:
+            # add the last 2 numbers in seq and append
+            seq.append(sum(seq[-2:]))
+
+
+def intsum():
```
Comment (mdgregor) [inline students/darrell/wk1 - assignments/generator_solution.py]: duplicate function?

File: students/darrell/wk1 - assignments/generator_solution.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    seq =[0]
+    while True:
+        if len(seq) == 1:
+            yield 0
+        else:
+            yield sum(seq)
+        seq.append(seq[-1] + 1)
+
+
+def doubler():
```
Comment (mdgregor) [inline students/darrell/wk1 - assignments/generator_solution.py]: duplicate function?

File: students/darrell/wk1 - assignments/generator_solution.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    n = 1
+    while True:
+        if n == 1:
+            yield 1
+        else:
+            yield n
+        n *= 2
+
+def fib():
```
Comment (mdgregor) [inline students/darrell/wk1 - assignments/generator_solution.py]: duplicate function?

**Comment by djh2020** [comment]
@mdgregor Not sure how the duplicates ended up in the file.  I cleaned it up and re-commited.  should be ok now.

**Pull request merged**